src/main/java/seedu/address/logic/commands/AddTagCommand.java

```diff
@@ -0,0 +1,205 @@
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof AddTagCommand)) {
+            return false;
+        }
+
+        AddTagCommand otherAddTagCommand = (AddTagCommand) other;
+        return index.equals(otherAddTagCommand.index)
+                && addTagDescriptor.equals(otherAddTagCommand.addTagDescriptor);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("index", index)
+                .add("editPersonDescriptor", addTagDescriptor)
+                .toString();
+    }
+
+    /**
+     * Stores the details to edit the person with. Each non-empty field value will replace the
+     * corresponding field value of the person.
+     */
+    public static class AddTagDescriptor {
+        private Name name;
+        private Phone phone;
+        private Email email;
+        private Set<Tag> tags;
+
+        public AddTagDescriptor() {}
+
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public AddTagDescriptor(AddTagCommand.AddTagDescriptor toAdd) {
+            setName(toAdd.name);
+            setPhone(toAdd.phone);
+            setEmail(toAdd.email);
+            setTags(toAdd.tags);
+        }
+
+        public void setName(Name name) {
+            this.name = name;
+        }
+
+        public Optional<Name> getName() {
+            return Optional.ofNullable(name);
+        }
+
+        public void setPhone(Phone phone) {
+            this.phone = phone;
+        }
+
+        public Optional<Phone> getPhone() {
+            return Optional.ofNullable(phone);
+        }
+
+        public void setEmail(Email email) {
+            this.email = email;
+        }
+
+        public Optional<Email> getEmail() {
+            return Optional.ofNullable(email);
+        }
+
+        /**
+         * Sets {@code tags} to this object's {@code tags}.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public void setTags(Set<Tag> tags) {
+            this.tags = (tags != null) ? new HashSet<>(tags) : null;
+        }
+
+        /**
+         * Returns an unmodifiable tag set, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code tags} is null.
+         */
+        public Optional<Set<Tag>> getTags() {
+            return (tags != null) ? Optional.of(Collections.unmodifiableSet(tags)) : Optional.empty();
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (other == this) {
+                return true;
+            }
+
+            // instanceof handles nulls
+            if (!(other instanceof AddTagCommand.AddTagDescriptor)) {
+                return false;
+            }
+
+            AddTagCommand.AddTagDescriptor otherAddTagDescriptor = (AddTagCommand.AddTagDescriptor) other;
```

Comment:
Will it be better to import the `AddTagDescriptor` instead of using the dot operator to directly access it? This will better align with the way we use `EditPersonDescriptor`.